apps/prairielearn/src/sprocs/ip_to_mode.sql

```diff
@@ -22,63 +20,66 @@ BEGIN
         mode := 'Public';
     END IF;
 
-    -- Does the user have an active PT reservation?
-    SELECT r.session_id
-    INTO v_session_id
-    FROM
-        pt_reservations AS r
-        JOIN pt_enrollments AS e ON (e.id = r.enrollment_id)
-    WHERE
-        e.user_id = ip_to_mode.authn_user_id
-        AND (date BETWEEN r.access_start AND r.access_end);
-
-    IF FOUND THEN
-        -- The user has a checked-in pt_reservation. Check whether the
-        -- reservation is in a testing center location that requires network
-        -- filtering.
-
-        SELECT loc.id, loc.filter_networks
-        INTO v_location_id, v_filter_networks
+    -- Consider each PT reservation which is either active or corresponds to
+    -- a session that will start soon or started recently.
+    FOR reservation IN
+        SELECT
+            r.session_id,
+            (ip_to_mode.date BETWEEN r.access_start AND access_end) AS reservation_active,
```

Comment:
A further enhancement we could make would be to also consider the `checked_in` time of a reservation and consider us to have `reservation_active = true` if `date BETWEEN checked_in AND checked_in + '1 hour'`, or something like that.

This would be helpful for course-run sessions where students are being checked in at the door as they enter the room, because they would then be in Exam mode as soon as they enter the room. This would also be helpful for center sessions that are have `location_filter_networks = false`.